### PR TITLE
Add authorization check options to iceberg doc

### PR DESCRIPTION
**Pull request by jhlodin**
Issue: https://github.com/trinodb/trino/issues/10548

### CODE REVIEW

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 1]
```diff
@@ -80,6 +80,36 @@ At a minimum, ``hive.metastore.uri`` must be configured:
     - Maximum number of partitions handled per writer.
     - 100
 
+.. _iceberg-authorization:
+
+Authorization checks
+^^^^^^^^^^^^^^^^^^^^
+
+You can enable authorization checks for the connector by setting
+the ``iceberg.security`` property in the catalog properties file. This
```
Comment (rosewms) [inline docs/src/main/sphinx/connector/iceberg.rst:89]: Is the 'iceberg.security' property in the catalog properties file by default or does the user need to add it and set the values?

If they have to add it to the file and choose a value, I suggest reframing the sentence to something like:

> To enable authorization checks, add the 'iceberg.security' property to the connector catalog properties file.

Comment (jhlodin) [reply docs/src/main/sphinx/connector/iceberg.rst:89]: I don't disagree, but we've used the wording "set/by setting the `example.property` property in the catalog properties file" just about everywhere else in the connector docs. This property is no different, I'd leave it as-is.

**Comment by mosabua** [comment]
@kokosing can we get your input here?

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 2]
```diff
@@ -80,6 +80,36 @@ At a minimum, ``hive.metastore.uri`` must be configured:
...
+    - The connector relies on system-level access control.
+  * - ``READ_ONLY``
+    - Operations that read data or metadata, such as :doc:`/sql/select` are
+      permitted. No operations that write data or metadata, such as
+      :doc:`/sql/create-table`, :doc:`/sql/insert`, or :doc:`/sql/delete` are
+      allowed.
+  * - ``FILE``
+    - Authorization checks are enforced using a configuration file whose path
+      is specified in the ``security.config-file`` catalog configuration
+      property. See :ref:`hive-file-based-authorization` for information on
```
Comment (kokosing) [inline docs/src/main/sphinx/connector/iceberg.rst:110]: It would be nice to move this out of `hive` as now it becomes more generic.

**Review by kokosing — APPROVED** [review]

File: docs/src/main/sphinx/connector/iceberg.rst (hunk 2 above)
Comment (jhlodin) [reply docs/src/main/sphinx/connector/iceberg.rst:110]: Agreed, but there's a lot of restructuring work that needs to be done to the Hive docs. There's already too many different docs covering these topics.

**Comment by jhlodin** [comment]
@kokosing Can you please merge this PR if all looks good? Thanks!

**Review by kokosing — APPROVED** [review]

**Comment by kokosing** [comment]
> Can you please merge this PR if all looks good? Thanks!

Sure. Please ping me once CI gets a green light.

File: docs/src/main/sphinx/connector/iceberg.rst (hunk 2 above)
Comment (jhlodin) [reply docs/src/main/sphinx/connector/iceberg.rst:110]: Followed up with https://github.com/trinodb/trino/pull/10800

**Pull request merged**